projects/shared/src/lib/contentlist/contentlist.component.ts

```diff
@@ -11,13 +12,15 @@ export class ContentlistComponent implements OnInit {
   @Input() contentList: Array<any> = [];
   @Output() contentSelect = new EventEmitter<any>();
 
+  constructor(private utilService: UtilsService) {}
+
   public contents: Array<ContentListItem>
   ngOnInit() {
 
     this.contents = this.contentList.map(content => {
-      const {color, backgroundColor} = this.getColors();
+      const {color, backgroundColor} = this.utilService.getColors();
```

Comment:
Think even further. Object mapping & conversion can be part of the Class rather than each component is doing. 

May be like content as class will expose the public methods convert. Pass JSON object as method parameter with will return the instance of the class with only class properties.


class Content {

public convert(content:<any>): IContent {
 var contentObj: <IContent>;
 // Object conversation logic
  return contentObj;
}

}


Implementation in component:
contentObj = new Content().convert(content)